examples/jetstream/api-migration/deno/main.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  AckPolicy,
+  connect,
+  consumerOpts,
+} from "https://deno.land/x/nats@v1.14.0/src/mod.ts";
+
+// Get the passed NATS_URL or fallback to the default. This can be
+// a comma-separated string. If not defined, it will default to localhost:4222
+const servers = Deno.env.get("NATS_URL")?.split(",");
+console.log(servers);
+
+// Create a client connection to an available NATS server.
+const nc = await (connect({ servers }));
+
+// Create a JetStream Manager context - this context has API we can use to
+// create streams and consumers.
+const jsm = await nc.jetstreamManager();
+// and create a stream
+await jsm.streams.add({
+  name: "EVENTS",
+  subjects: ["events.>"],
+});
+
+// We also create a JetStream context that provides a mechanism
+// for publishing data to JetStream and to create "subscriptions"
+// on a consumer:
+const js = nc.jetstream();
+// Here we add 20 messages to a stream
+const proms = Array.from({ length: 20 })
+  .map((_v, idx) => {
+    return js.publish(`events.${idx}`);
+  });
+await Promise.all(proms);
+
+// Processing messages using Legacy JetStream is fairly cumbersome
+// and because the JetStream subscription could create or update consumers
+// under the covers, it leads to some complications. Specially if you had
+// more than a single process that examined a stream. You could get unintended
+// consumer updates, etc.
+
+// # Legacy Processing a Stream
+//
+// Using the legacy API, the easiest way to continuously receive messages
+// is to use push consumers.
+// The `subscribe()` API call was intended for these "push" consumers. These
+// look very natural to NATS users.
+let opts = consumerOpts()
+  .deliverTo("eventprocessing")
+  .ackExplicit()
+  .manualAck();
+
+const pushSub = await js.subscribe("events.>", opts);
+for await (const m of pushSub) {
+  console.log(`legacy push subscriber received ${m.subject}`);
+  m.ack();
+  // we can check if the stream has more messages
+  // by checking the number of pending messages, and break
+  // if we are done.
+  if (m.info.pending === 0) {
+    // breaking from the iterator will stop the subscription
```

Comment:
This does not render nicely:

<img width="885" alt="image" src="https://github.com/ConnectEverything/nats-by-example/assets/15184508/3ff46480-f6ce-4d8e-a4b4-4d5904e0c4b8">